src/entity/notifications/NotificationPreference.ts

```diff
@@ -0,0 +1,28 @@
+import { Column, Entity, PrimaryColumn, TableInheritance } from 'typeorm';
+import {
+  NotificationPreferenceStatus,
+  NotificationPreferenceType,
+  NotificationType,
+} from '../../notifications/common';
+
+@Entity()
+@TableInheritance({ column: { type: 'varchar', name: 'type' } })
+export class NotificationPreference {
+  @PrimaryColumn({ type: 'text' })
+  uniqueKey: string;
```

Comment:
Would this not work as simply using existing naming for `id`?
Or what do you plan to merge as the uniqueKey?

Reply:
Yes, since we will have the constraint on the dedicated child entities. This is so we will have the uniqueness of the row as adding new primary keys (new child entities) over time will have an additional overhead resource-wise. 

> I assumed at first we just have a generated ID as it adds no value correct?

Yes, it holds no "value" in terms of definition, either generated id or using the reference id should work.

Reply:
@rebelchris merging this in a few moments. Let me know if you have further concerns 🥳

Reply:
Fine if everyone else agrees :)
Not really blocking, just a stickler for naming that get's reused 🙏

Reply:
I am fine with `referenceId`

Reply:
Fine as well